web_dms_media/models/dms.py

```diff
@@ -0,0 +1,53 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from werkzeug.urls import url_quote
+
+from odoo import api, models, fields, tools
+
+SUPPORTED_IMAGE_MIMETYPES = [
+    'image/gif', 'image/jpe', 'image/jpeg', 'image/jpg', 'image/gif', 'image/png', 'image/svg+xml'
+]
+
+
+class DMSFile(models.Model):
+    _inherit = "dms.file"
+
+    local_url = fields.Char("Attachment URL", compute='_compute_local_url')
+    image_src = fields.Char(compute='_compute_image_src')
+    image_width = fields.Integer(compute='_compute_image_size')
+    image_height = fields.Integer(compute='_compute_image_size')
+    original_id = fields.Many2one('dms.file', string="Original (unoptimized, unresized) attachment", index=True)
+
+    def _compute_local_url(self):
+        for attachment in self:
+            attachment.local_url = '/web/image/dms.file/%s/image_1024' % attachment.id
```

Comment:
I prefer this method of string formatting as it the current way of doing it, the old syntax is deprecated.
f'/web/image/dms.file/{attachment.id}/image_1024'